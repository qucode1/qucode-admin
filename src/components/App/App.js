import React from 'react'
import {
  BrowserRouter as Router
} from 'react-router-dom'

import Header from '../Header/Header'
import Sidebar from '../Sidebar/Sidebar'
import Routes from '../Routes/Routes'

const App = () => (
  <Router>
    <div>
      <Header />
      <div className='container'>
        <Sidebar />
        <main>
          <Routes />
        </main>
        <style jsx global>{`
          html, body {
            box-sizing: border-box
          }
          html, body, body > div, body > div > div {
            height: 100%;
            width: 100%;
          }
        `}</style>
        <style jsx>{`
          .container {
            display: flex;
            height: calc(100% - 70px);
            padding-top: 70px
          }
          main {
            flex: 3
          }
        `}</style>
      </div>
    </div>
  </Router>
)
export default App
