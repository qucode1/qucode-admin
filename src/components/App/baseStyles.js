export const baseStyles = `
  :root{
    --site-width: 98rem
  }

  html {
    box-sizing: border-box;
    font-size: 62.5%;
    font-family: sans-serif;
    margin: 0;
    padding: 0;
  }
  html, body, body > div:first-of-type, body > div:first-of-type > div:first-of-type {
    height: 100%;
    width: 100%;
  }
  body {
    font-size: 1.5rem;
    margin: 0;
  }
  ul {
    list-style: none;
    padding: 0;
    margin: 0
  }
  a {
    text-decoration: none;
  }
`
