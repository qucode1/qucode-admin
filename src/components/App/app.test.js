import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import App from './App'

describe('<App />', () => {
  it('App component should render as expected', () => {
    const wrapper = shallow(<App />)
    const tree = toJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
})
