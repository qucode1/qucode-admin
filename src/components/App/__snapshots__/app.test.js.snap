// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App component should render as expected 1`] = `
<BrowserRouter>
  <div>
    <Header />
    <div
      className="container"
    >
      <Routes />
    </div>
  </div>
</BrowserRouter>
`;
